tests/main/views/test_direct_award.py

```diff
@@ -0,0 +1,499 @@
+import json
+from datetime import datetime
+import math
+import random
+
+import pytest
+from tests.helpers import FixtureMixin
+from tests.bases import BaseApplicationTest
+
+from dmapiclient.audit import AuditTypes
+from app.models import DATETIME_FORMAT, DirectAwardProjectUser, AuditEvent, DirectAwardProject, User, DirectAwardSearch
+from ...helpers import DIRECT_AWARD_SEARCH_URL, DIRECT_AWARD_PROJECT_NAME
+
+
+class DirectAwardSetupAndTeardown(BaseApplicationTest, FixtureMixin):
+    def setup(self):
+        super(DirectAwardSetupAndTeardown, self).setup()
+        self.user_id = self.setup_dummy_user(id=1, role='buyer')
+        self.direct_award_project_name = DIRECT_AWARD_PROJECT_NAME
+        self.direct_award_search_url = DIRECT_AWARD_SEARCH_URL
+
+    def teardown(self):
+        super(DirectAwardSetupAndTeardown, self).teardown()
+
+
+class TestDirectAwardListProjects(DirectAwardSetupAndTeardown):
+    def setup(self):
+        super(TestDirectAwardListProjects, self).setup()
+        self.project_id = self.create_direct_award_project(user_id=self.user_id,
+                                                           project_name=self.direct_award_project_name)
+
+    def test_list_projects_400s_if_no_user_id(self):
+        res = self.client.get('/direct-award/projects')
+        assert res.status_code == 400
+
+    def test_list_projects_200s_with_user_id(self):
+        res = self.client.get('/direct-award/projects?user_id={}'.format(self.user_id))
+        assert res.status_code == 200
+
+    def test_list_projects_returns_all_projects_for_user(self):
+        self.create_direct_award_project(user_id=self.user_id, project_id=self.project_id + 1)
+
+        # Create a project for another user, i.e. one that shouldn't be returned
+        self.setup_dummy_user(id=self.user_id + 1, role='buyer')
+        self.create_direct_award_project(user_id=self.user_id + 1, project_id=self.project_id + 2)
+
+        res = self.client.get('/direct-award/projects?user_id={}'.format(self.user_id))
+        data = json.loads(res.get_data(as_text=True))
+
+        assert res.status_code == 200
+        assert len(data['projects']) == 2
+
+        with self.app.app_context():
+            projects = DirectAwardProject.query.filter(DirectAwardProject.users.any(User.id == self.user_id))
+
+            assert all([x.serialize() in data['projects'] for x in projects.all()])
```

Comment:
I like x, it's one of my favourite chars, but maybe we could have something a bit more descriptive? :P